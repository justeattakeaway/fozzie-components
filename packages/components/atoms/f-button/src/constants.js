// eslint-disable-next-line import/prefer-default-export
export const VALID_BUTTON_TYPES = {
    iconButton: ['primary', 'secondary', 'ghost', 'ghostTertiary'],
    button: ['primary', 'secondary', 'outline', 'ghost', 'link']
};

export const VALID_BUTTON_SIZES = ['xsmall', 'small', 'medium', 'large'];
