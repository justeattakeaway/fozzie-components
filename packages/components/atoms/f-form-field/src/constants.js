export const VALID_INPUT_TYPES = ['text', 'email', 'password', 'radio', 'checkbox', 'number'];
export const CUSTOM_INPUT_TYPES = ['dropdown'];
export const DEFAULT_INPUT_TYPE = 'text';

export const VALID_LABEL_STYLES = ['default', 'inline', 'inlineNarrow'];

export const MOBILE_WIDTH = 768;
