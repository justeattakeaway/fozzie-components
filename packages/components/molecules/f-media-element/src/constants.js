export const ALIGN = {
    LEFT: 'left',
    RIGHT: 'right',
    CENTER: 'center'
};

export const FONT_SIZE = {
    SM: 'sm',
    MD: 'md',
    LG: 'lg',
    XL: 'xl',
    XXL: 'xxl'
};

export const MODIFIER_RULES_MAP = {
    '>=': 'gte',
    '>': 'gt',
    '<': 'lt',
    '<=': 'lte'
};

export const MODIFIER_OPPOSITES_RULES_MAP = {
    '>=': 'lt',
    '>': 'lte',
    '<': 'gte',
    '<=': 'gt'
};
