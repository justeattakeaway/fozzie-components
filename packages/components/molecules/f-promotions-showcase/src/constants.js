export const STYLE_TEXT = 'styleText';
export const STYLE_EMPHASIZED = 'styleEmphasized';
export const STYLE_LINK = 'styleLink';

export default {
    STYLE_EMPHASIZED,
    STYLE_LINK,
    STYLE_TEXT
};
