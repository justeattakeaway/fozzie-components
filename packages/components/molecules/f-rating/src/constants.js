// eslint-disable-next-line import/prefer-default-export
export const VALID_STAR_RATING_SIZES = ['xsmall', 'small', 'medium', 'large'];
export const VALID_STAR_RATING_DISPLAY_TYPE = ['short', 'medium', 'long'];
export const VALID_STAR_FONT_SIZES = ['default', 'large'];
