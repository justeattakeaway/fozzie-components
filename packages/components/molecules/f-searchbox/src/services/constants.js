export const LOCATION_COOKIE_PROPS = [
    'unitNumber',
    'street',
    'houseNo',
    'city',
    'sublocality',
    'state'
];
export const JE_LOCATION = 'je-location';

export const JE_FULL_ADDRESS_DETAILS = 'je-full-address-details';

export const VUEX_MODULE_NAMESPACE = 'fSearchboxModule';
