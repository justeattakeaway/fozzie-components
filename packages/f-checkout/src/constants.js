export const CHECKOUT_METHOD_COLLECTION = 'Collection';
export const CHECKOUT_METHOD_DELIVERY = 'Delivery';

export const VALID_CHECKOUT_METHOD = [CHECKOUT_METHOD_COLLECTION, CHECKOUT_METHOD_DELIVERY];
