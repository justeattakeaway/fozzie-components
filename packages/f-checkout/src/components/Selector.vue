<template>
    <div
        data-test-id="form-select"
        :class="[
            $style['o-form-select'],
            (selectedTime ? $style['o-form-select--float'] : '')
        ]">
        <label
            for="time-selection"
            data-test-id="fulfillment-time-label"
            :class="$style['o-form-select-label']">
            {{ orderMethod }}
        </label>
        <select
            id="time-selection"
            v-model="selectedTime"
            :class="$style['o-form-select-input']"
            data-test-id="fulfillment-time"
            @change="selectionChanged">
            <option
                v-for="(time, index) in fulfillment.times"
                :key="index"
                :value="time.from">
                {{ time.label.text }}
            </option>
        </select>
    </div>
</template>

<script>
import { mapState } from 'vuex';
import { CHECKOUT_METHOD_DELIVERY } from '../constants';

export default {
    data () {
        return {
            selectedTime: null
        };
    },

    computed: {
        ...mapState('checkout', [
            'fulfillment',
            'serviceType'
        ]),

        orderMethod () {
            return this.serviceType === CHECKOUT_METHOD_DELIVERY
                ? this.$t('labels.deliveryOrderMethod')
                : this.$t('labels.collectionOrderMethod');
        }
    },

    watch: {
        'fulfillment.times' (newValue) {
            const selected = newValue.find(t => t.selected);
            if (selected) {
                this.selectedTime = selected.from;
            }
        }
    },

    methods: {
        selectionChanged () {
            this.fulfillment.times.forEach(el => { el.selected = false; });
            const newSelected = this.fulfillment.times.find(t => t.from === this.selectedTime);
            if (newSelected) {
                newSelected.selected = true;
            }
        }
    }
};
</script>

<style lang="scss" module>
$form-label-colour                        : $grey--dark;
$form-input-colour                        : $color-text;
$form-input-bg                            : $white;
$form-input-borderRadius                  : 3px;
$form-input-borderWidth                   : 1px;
$form-input-borderColour                  : $grey--light;
$form-input-borderColour--focus           : $grey--dark;

.o-form-select {
    position: relative;
    height: 60px;
    margin: spacing(x2) 0;
    padding: 0;
    @include font-size(body-l);
    font-family: $font-family-base;
    color: $form-input-colour;
    font-weight: $font-weight-base;
    background-color: $form-input-bg;
    border: $form-input-borderWidth solid $form-input-borderColour;
    border-radius: $form-input-borderRadius;

    .o-form-select-label {
        display: block;
        position: absolute;
        top: 50%;
        transform: translateY(-50%);
        padding: spacing();
        color: $form-label-colour;
        cursor: pointer;
    }

    .o-form-select-input {
        height: 100%;
        width: 100%;
        padding: spacing(x0.5);
        border: none;
        cursor: pointer;
        color: $color-text;
    }
}

/**
 * Modifier â€“ .o-form-select--float
 *
 * Moves the select label to sit above chosen option
 */
.o-form-select--float {
    .o-form-select-label {
        @include font-size(body-s);
        top: 15px;
    }

    .o-form-select-input {
        padding-top: spacing(x3);
    }
}
</style>
