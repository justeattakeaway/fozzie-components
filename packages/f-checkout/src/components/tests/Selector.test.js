import Vuex from 'vuex';
import { VueI18n } from '@justeat/f-globalisation';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import { CHECKOUT_METHOD_DELIVERY, CHECKOUT_METHOD_COLLECTION } from '../../constants';
import Selector from '../Selector.vue';
import tenantConfigs from '../../tenants';

const localVue = createLocalVue();

localVue.use(VueI18n);
localVue.use(Vuex);

const defaultState = {
    id: '',
    serviceType: CHECKOUT_METHOD_DELIVERY,
    customer: {
        firstName: 'John',
        mobileNumber: '447111111111'
    },
    fulfillment: {
        times: [],
        address: {
            line1: '1 Bristol Road',
            line2: 'Flat 1',
            city: 'Bristol',
            postcode: 'BS1 1AA'
        }
    },
    notes: [],
    isFulfillable: true,
    notices: [],
    messages: []
};

const defaultActions = {
    getCheckout: jest.fn(),
    postCheckout: jest.fn()
};

const i18n = {
    locale: 'en-GB',
    messages: tenantConfigs['en-GB']
};

const createStore = (state = defaultState, actions = defaultActions) => new Vuex.Store({
    modules: {
        checkout: {
            namespaced: true,
            state,
            actions
        }
    },
    hasModule: jest.fn(() => true)
});

describe('Selector', () => {
    allure.feature('Checkout-Selector');
    const propsData = {};

    it('should be defined', () => {
        // Arrange & Act
        const wrapper = shallowMount(Selector, {
            store: createStore(),
            i18n,
            localVue,
            propsData
        });

        // Assert
        expect(wrapper.exists()).toBe(true);
    });

    it('should show the delivery label when the `serviceType` is `delivery`', async () => {
        // Arrange & Act
        const wrapper = shallowMount(Selector, {
            store: createStore({ ...defaultState, serviceType: CHECKOUT_METHOD_DELIVERY }),
            i18n,
            localVue,
            propsData
        });

        const selectorLabel = wrapper.find('[data-test-id="fulfillment-time-label"]');

        // Assert
        expect(selectorLabel.html()).toMatchSnapshot();
    });

    it('should show the collection label when the `serviceType` is `collection`', async () => {
        // Arrange & Act
        const wrapper = shallowMount(Selector, {
            store: createStore({ ...defaultState, serviceType: CHECKOUT_METHOD_COLLECTION }),
            i18n,
            localVue,
            propsData
        });

        const selectorLabel = wrapper.find('[data-test-id="fulfillment-time-label"]');

        // Assert
        expect(selectorLabel.html()).toMatchSnapshot();
    });

    describe('data ::', () => {
        describe('selectedTime ::', () => {
            // TODO: Component is working correctly but the tests do not work with `scss modules`

            xit('should add class to display label above option when not null', async () => {
                // Arrange
                const data = { selectedTime: 'testTime' };
                const wrapper = shallowMount(Selector, {
                    store: createStore(),
                    i18n,
                    localVue,
                    propsData
                });

                const selector = wrapper.find("[data-test-id='form-select']");

                // Act
                wrapper.setData(data);
                await wrapper.vm.$nextTick();

                // Assert
                expect(wrapper.html()).toContain('o-form-select--float');
                expect(selector.classes()).toContain('o-form-select--float');
            });

            it('should remove class to display label centrally when null', () => {
                // Arrange & Act
                const wrapper = shallowMount(Selector, {
                    store: createStore(),
                    i18n,
                    localVue,
                    propsData
                });

                const selector = wrapper.find("[data-test-id='form-select']");

                // Assert
                expect(selector.classes()).not.toContain('o-form-select--float');
            });
        });
    });
});
