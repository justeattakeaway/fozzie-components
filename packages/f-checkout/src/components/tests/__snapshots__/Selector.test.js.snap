// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector should show the collection label when the \`serviceType\` is \`collection\` 1`] = `
<label for="time-selection" data-test-id="fulfillment-time-label">
  labels.collectionOrderMethod
</label>
`;

exports[`Selector should show the delivery label when the \`serviceType\` is \`delivery\` 1`] = `
<label for="time-selection" data-test-id="fulfillment-time-label">
  labels.deliveryOrderMethod
</label>
`;
